app/models/place.rb

```diff
@@ -1,3 +1,18 @@
 class Place < ActiveRecord::Base
-  belongs_to :user, foreign_key: :admin_id
+
+  belongs_to :admin, class_name: 'User'
+  has_many :keys
+  has_many :logs, through: :keys
+
+  def available_keys
+    current_time = Time.now
+    all_keys = keys.where([" start_date <= ? and end_date >= ? ",current_time,current_time])
+    all_keys << ClientKey.find_by(client_id: 0).key
+    all_keys
+  end
+
+  def default
+    current_user.places.where(default_place: true)
```

Comment:
Does this model have access to current_user? Possibly not. This should maybe be something more like
```
def self.default_for_user(user)
  where(default_place: true).where(user: user)
end
```

BUT is it actually used anywhere?